src/test/java/com/google/sps/servlets/MatchesListServletTest.java

```diff
@@ -0,0 +1,174 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.json.JSONObject;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.List;
+import java.util.Arrays;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Before;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableList;
+import com.google.gson.Gson;
+
+@RunWith(JUnit4.class)
+public class MatchesListServletTest {
+  private static final String MATCH_INFO_ENTITY = "match-info";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String POTENTIAL_MATCHES_PROPERTY = "potential-matches";
+  private static final String FRIENDED_IDS_PROPERTY = "friended-ids";
+  private static final String PASSED_IDS_PROPERTY = "passed-ids";
+  private static final String MATCHES_LIST_PROPERTY = "matches-list";
+  private static final String USER_ID_REQUEST_URL_PARAM = "userid";
+
+  //Test user ids
+  private static final String TEST_USER_ID = "5555";
+  private static final String TEST_CONNECTION_1_ID = "1776";
+  private static final String TEST_CONNECTION_2_ID = "1234";
+  private static final String TEST_CONNECTION_3_ID = "9876";
+
+  private static final List<String> EMPTY_LIST = Arrays.asList();
+
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock
+  private HttpServletRequest mockRequest;
+ 
+  @Mock
+  private HttpServletResponse mockResponse;
+
+  private MatchesListServlet servletUnderTest;
+  private DatastoreService datastore;
+  private StringWriter responseWriter;
+  private PrintWriter writer;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+
+    helper.setUp();
+
+    servletUnderTest = new MatchesListServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests the case in which a user's match list is empty.
+   *
+   * <p>Should return an empty list.
+   */
+  @Test
+  public void matchInfoWithEmptyList() throws IOException{
+    addMatchInfoEntityToDatastore(TEST_USER_ID, ImmutableList.of(),
```

Comment:
While I would *really really* like a builder here, it's overkill.

That said, we do need more context here on what these empty lists are. We achieve that this way:

```
addMatchInfoEntityToDatastore(TEST_USER_ID, /* param1Name= */ ImmutableList.of(), /* param2Name= */ ImmutableList.of(),....
```

User id is pretty clear here, but you need to label the other method parameters - they are just lists w/o inline context. Also, make sure you need all of the lists. I think I see at least the first one is always empty. If you're not setting it in the test, just set it to a default it in the `addMatchInfoEntityToDatastore` method itself.

Reply:
Got it! I saw that the passed and potential match lists were always empty so I removed those parameters from addMatchInfoEntityToDatastore. And I kept the friendedUsers lists because I realized that users on the match list should also be in the friendedUsers list.